application/main.py

```diff
@@ -573,7 +572,92 @@ def __eq__(self, other):
                (self.validation_results == other.validation_results)
 
 
+@application.route('/build_implementer_agreement_results/<id>')
+@login_required
+def build_implementer_agreement_results(user_data, id):
+    with database.Session() as session:
+
+        model = session.query(database.model).filter(database.model.code == id).all()[0]
+        tasks = {t.task_type: t for t in model.tasks}
+        instances = {i: i for i in model.instances}
+
+        imp_agreement_results = [imp_ag.serialize() for imp_ag in tasks['implementer_agreements_task'].results]
+        instances_results = [ins.serialize() for ins in instances]
+
+        then_statements = []
+        unique_steps = set()
+
+        for imp_ag in tasks['implementer_agreements_task'].results:
+            imp_ag = imp_ag.serialize()
+            unique_steps.add(imp_ag['step'])
```

Comment:
Can't this be written simply as : 
`unique_steps = set(result['step'] for result in imp_agreement_results)` ?

Reply:
Done.